eng/versioning/external_dependencies.txt

```diff
@@ -41,6 +41,8 @@ org.apache.maven.plugins:maven-source-plugin
 org.apache.maven.plugins:maven-surefire-plugin
 org.codehaus.mojo:exec-maven-plugin
 
-org.mockito:mockito-core;4.11.0
```

Comment:
why remove `4.11.0`?

Reply:
When syncing, we always take version from SDK repo e.g. https://github.com/Azure/azure-sdk-for-java/blob/main/eng/versioning/external_dependencies.txt

So we don't have version here in this repo.